guides/release/state-management/computed-properties.md

```diff
@@ -1 +1,634 @@
+  }
+
+  @computed('firstName', 'lastName')
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+
+let ironMan = new Person('Tony', 'Stark');
+
+ironMan.fullName; // "Tony Stark"
+```
+
+This computed property works just like a normal getter/setter, with two key
+differences:
+
+1. It will cache its value by default, and it will only update that value if its
+   _dependencies_, in this case the `firstName` and `lastName` properties,
+   change.
+
+   ```javascript
+   class Counter {
+     _count = 0;
+
+     @computed
+     get count() {
+       console.log('counted!');
+       return this._count;
+     }
+   }
+
+   let counter = new Counter();
+
+   counter.count; // logs 'counted!'
+   counter.count; // logs nothing, the values was cached and hasn't updated
+   ```
+
+2. It will notify other "watchers", such as other computed properties and
+   templates, if any of its dependencies has updated and it needs to be
+   recalculated.
+
+### Specifying Dependencies
+
+So far we've seen computed properties with dependencies on properties that are
+_local_ to the object, but you can specify a few other types of dependencies:
+
+- **Chain dependencies.** If you need to specify a dependency on an _object_,
+  you can use dot notation to do so:
+
+  ```js
+  class Profile {
+    constructor(user) {
+      set(this, 'user', user);
+    }
+
+    @computed('user.firstName', 'user.lastName')
+    get userName() {
+      return `${this.user.firstName} ${this.user.lastName}`;
+    }
+  }
+  ```
+
+  When doing this for more than one value on the object, you can also use a
+  special truncated syntax as shorthand:
+
+  ```js
+  class Profile {
+    constructor(user) {
+      set(this, 'user', user);
+    }
+
+    @computed('user.{firstName,lastName}')
+    get userName() {
+      return `${this.user.firstName} ${this.user.lastName}`;
+    }
+  }
+  ```
+
+  Note that no spaces are allowed in this truncated syntax, Ember will assert if
+  you place any inside of it.
+
+- **Array dependencies.** It's possible to depend on an array, and the items in
+  the array, by watching the `[]` property on the array:
+
+  ```js
+  class Person {
+    constructor(friends = []) {
+      set(this, 'friends', friends);
+    }
+
+    @computed('friends.[]')
+    get friendNames() {
+      return this.friends.map(friend => friend.name);
+    }
+  }
+  ```
+
+  You can also depend directly on a _property_ of each item in the array using
+  `@each` syntax:
+
+  ```js
+  class Person {
+    constructor(friends = []) {
+      set(this, 'friends', friends);
+    }
+
+    @computed('friends.@each.name')
+    get friendNames() {
+      return this.friends.map(friend => friend.name);
+    }
+  }
+  ```
+
+  However, you cannot _chain_ on these properties, as it is a performance
+  pitfall. You can only do 1 level of `@each` watching.
+
+### Defining Setters
+
+If you define a setter for your computed property, it'll work just like a normal
```

Comment:
```suggestion
If you define a setter for your computed property, it will work just like a normal
```

Feedback from non-native-English-speakers is that contractions can be difficult. We do use `let's` liberally and I don't usually remove it.